web/module/lexicon.php

```diff
@@ -69,7 +73,21 @@ function match_entry($key) {
 	function match_page($key, $entry_name) {
 		return match("botania\.page\.$entry_name\d+", $key);
 	}
-	function quote_keys($key) {
-		return "/\\$\\(k:$key\\)/";
+	function key_resolve($match) {
+		global $default_keys; // i'm sorry, i don't know php
+		return (array_key_exists($match[1], $default_keys)
+			? $default_keys[$match[1]]
+			: "UNKNOWN KEY");
```

Comment:
where is this default_keys array being filled? can it just be passed in instead of a global?

Reply:
It's filled on line 17-- it's a lookup table for `$(k:) codes`; no, because `key_resolve` has to take one array and return a string where it's used in preg_replace_callback, and it's more performant to keep it at filescope.